src/Playwright/Program.cs

```diff
@@ -0,0 +1,123 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Microsoft.Playwright.Helpers;
+
+namespace Microsoft.Playwright
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var p = new Program();
+            p.Run(args);
+        }
+
+        public void Run(string[] args)
+        {
+            string path = Directory.GetCurrentDirectory();
+
+            if (args != null && args.Length > 1 && args[0] == "-p")
+            {
+                path = Path.Combine(Directory.GetCurrentDirectory(), args[1]);
+                var newArgs = new string[args.Length - 2];
+                for (var i = 0; i < args.Length; ++i)
+                {
+                    newArgs[i] = args[i + 2];
+                }
+
+                args = newArgs;
+            }
+
+            string pwPath = null;
+            try
+            {
+                pwPath = Paths.GetExecutablePath();
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Please install Playwright:");
+                Console.WriteLine("   dotnet add package Microsoft.Playwright");
+                return;
+            }
+
+            var playwrightStartInfo = new ProcessStartInfo(pwPath, string.Join(" ", args))
+            {
+                UseShellExecute = false,
+                CreateNoWindow = true,
+                RedirectStandardError = true,
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+            };
+
+            using var pwProcess = new Process()
+            {
+                StartInfo = playwrightStartInfo,
+            };
+
+            playwrightStartInfo.EnvironmentVariables.Add("PW_CLI_TARGET_LANG", "csharp");
+            playwrightStartInfo.EnvironmentVariables.Add("PW_CLI_NAME ", "playwright");
+
+            using var outputWaitHandle = new AutoResetEvent(false);
+            using var errorWaitHandle = new AutoResetEvent(false);
+
+            pwProcess.OutputDataReceived += (_, e) =>
+            {
+                if (e.Data == null)
+                {
+                    outputWaitHandle.Set();
+                }
+                else
+                {
+                    Console.WriteLine(e.Data);
+                }
+            };
+
+            pwProcess.ErrorDataReceived += (_, e) =>
+            {
+                if (e.Data == null)
+                {
+                    errorWaitHandle.Set();
+                }
+                else
+                {
+                    Console.Error.WriteLine(e.Data);
+                }
+            };
+
+            pwProcess.Start();
+
+            pwProcess.BeginOutputReadLine();
+            pwProcess.BeginErrorReadLine();
+
+            pwProcess.WaitForExit();
+            outputWaitHandle.WaitOne(5000);
```

Comment:
I keep wondering about these 5000...

Reply:
Oh, I did not look at the code itself, I assumed it just worked. Let me follow up on this :)